src/main.cpp

```diff
@@ -30,9 +75,29 @@
  */
 int main(int argc,char **argv)
 {
-  // do a check, if CHM HTML WebView is open ...
+  /**
+    * \brief code added by paule32, to check for chm viewer instances under
+    *        MS-Windows 10, 11 is open ...
+    */
   #if defined(_WIN32) || defined(_WIN64)
-  
+  const TCHAR *processName = _T("msedgewebview2.exe");
+  DWORD processId = 0;
+  if (IsProcessRunning(processName, &processId)))) {
+      int response = MessageBox(0,
+      _T("The HTML Viewer is already open.\nWould you exit, or close the viewer ?"),
+      _T("Warning"),
+      MB_ICONQUESTION | MB_OKCANCEL);
```

Comment:
My 2 cents
- Why should doxygen test whether or not a viewer is running and even without looking if a chm file is generated?
- popping up a message box in the console application doxygen is completely not done.

Reply:
Because, when you would like to generate a CHM HTML Project for using the Microsoft HTML Workshop Compiler, to produce WebView1/2 Windows compiled html files - for e.g. product documentation, Doxygen takes no care, if the WebViewer is already open, yet.
When it is open, you can not generate a CHM File with hhc.exe - or it is empty.

The popout of the MessageBox can be re-design, so you get a message in the message.log file ?
It was a on-the-fly idea from me, that could be improof, indeed.